packages/tools/kadena-cli/src/devnet/commands/simulate.ts

```diff
@@ -16,10 +16,13 @@ export const simulateCommand: CreateCommandReturnType = createCommand(
     globalOptions.simulateTokenPool(),
     globalOptions.simulateMaxAmount(),
     globalOptions.simulateSeed(),
+    globalOptions.simulateMaxTime({ isOptional: true }),
   ],
   async (config) => {
```

Comment:
With the changes made by bart in the PR we were waiting for , that was merged before you started.

you see that every command has
` async (config) => {`
- some logic for getting/passing config coming from Config
- a action function `result = await someAction(config, whatever)`
```
export const someAction = async (config, whatever): Promise<CommandResult<{}>> => {
   try {
    // something
    return { success: true, data: response };
  } catch (error) {
    return {
      success: false,
      errors: [`Error in someAction: ${error.message}`],
    };
  }
};
```

This so someAction can be tested independent

- a asserter: `assertCommandError(result)`
- some login,

Reply:
hmmm on second thought  since there is no actual result this might not actually be needed, there is nothing to assert if its non-existent :P

Reply:
it can still have errors right? Like `guardDocker` can throw an error, you want to have a warning if container is not running. That could still use the warnings/errors messages of the command-result structure, even if no data is returned

Reply:
Yes or even on the simulation itself, we can have errors. I will implement some error handling